src/amo/components/UserProfileEdit/index.js

```diff
@@ -210,11 +216,37 @@ export class UserProfileEditBase extends React.Component<Props, State> {
       hasEditPermission,
       i18n,
       isUpdating,
+      router,
       user,
       username,
     } = this.props;
 
-    if (!currentUser || (currentUser && user && !hasEditPermission)) {
+    if (!currentUser) {
+      return (
+        <div className="UserProfileEdit">
+          <Card className="UserProfileEdit-user-links">
+            <AuthenticateButton
+              noIcon
+              location={router.location}
+              logInText={i18n.gettext('Log in to continue')}
```

Comment:
I saw your comment about why this is just `Log in to continue`, but I wonder if we could come up with something a bit more meaningful. I'm not sure we need to worry about the accuracy of the button's message in the case where a user is going to get a 404, because they shouldn't be trying to access that page anyway. So maybe `Log in to edit the profile`?